src/_zkapauthorizer/model.py

```diff
@@ -296,7 +287,7 @@ def snapshot(self) -> bytes:
         Create and return a consistent, self-contained snapshot of the underlying
         database state.
         """
-        return self._connection.snapshot()
+        return snapshot(self._connection)
```

Comment:
since it doesn't make sense for the abstract Connection protocol to have a snapshot method (the real sqlite3 Connection doesn't) I made this method into a function.  I could have annotated `self._connection` as `_ReplicationCapableConnection` instead but I didn't want to depend on that type in `model.py`, and also I got it working this way first and didn't want to try it another way.